server/routes/prisonIncentiveLevels.ts

```diff
@@ -483,13 +483,19 @@ async function ensureDefaultLevelExists(
   incentivesApi: IncentivesApi,
   prisonId: string,
   prisonIncentiveLevels: PrisonIncentiveLevel[],
-  defaultLevelCode = 'STD',
 ) {
   if (!prisonIncentiveLevels.some(prisonIncentiveLevel => prisonIncentiveLevel.defaultOnAdmission)) {
+    logger.warn(`${prisonId} is missing a default level`)
+    const defaultLevelCode =
+      (await incentivesApi.getPrisonIncentiveLevels(prisonId, true))
+        .filter(prisonIncentiveLevel => prisonIncentiveLevel.defaultOnAdmission)
+        .map(prisonIncentiveLevel => prisonIncentiveLevel.levelCode)?.[0] ?? 'STD'
```

Comment:
What do you think about using `.find()` instead?

I haven't tested this so it may be wrong but you should be able to write something along the lines of

```
.find(prisonIncentiveLevel => prisonIncentiveLevel.defaultOnAdmission)?.levelCode ?? 'STD'
```

which I think reads a bit better, especially as you wouldn't have to use the `map` and the `.[0]` indexing.

Reply:
oops i forgot that was a method!

Reply:
I like it when the intend is to find one thing (like in this case where you're looking for the "default on admission level") - and it tends to make code nicer in these occurences :)

Reply:
totes agree